pkg/providers/imagefamily/nodeimage_test.go

```diff
@@ -68,15 +68,32 @@ func getExpectedTestCIGImages(imageFamily string, version string, kubernetesVers
 }
 
 //nolint:unparam // might always be using the same version in test, but could change in the future
-func getExpectedTestSIGImages(imageFamily string, version string, kubernetesVersion string) []imagefamily.NodeImage {
+func getExpectedTestSIGImages(imageFamily string, fipsMode string, version string, kubernetesVersion string) []imagefamily.NodeImage {
 	var images []imagefamilytypes.DefaultImageOutput
-	if imageFamily == v1beta1.Ubuntu2204ImageFamily {
-		images = imagefamily.Ubuntu2204{}.DefaultImages()
-	} else if imageFamily == v1beta1.AzureLinuxImageFamily {
-		if imagefamily.UseAzureLinux3(kubernetesVersion) {
-			images = imagefamily.AzureLinux3{}.DefaultImages()
-		} else {
-			images = imagefamily.AzureLinux{}.DefaultImages()
+	if fipsMode == v1beta1.FIPSEnabled {
```

Comment:
nit: I think you could refactor this a bit, to have the check for `fipsMode` happen at the end,a nd have the internal if cases give you the image family, not the image list, and then simply call `FIPSImages` or `DefaultImages` afterwards based on fipsMode.

Beyond that you'd just need to have a check for `fipsMode` inside the `UbuntuImageFamily` branch for selecting the image family.